src/decode.c

```diff
@@ -402,22 +402,27 @@ void PacketBypassCallback(Packet *p)
 {
     /* Don't try to bypass if flow is already out or
      * if we have failed to do it once */
-    int state = SC_ATOMIC_GET(p->flow->flow_state);
-    if ((state == FLOW_STATE_LOCAL_BYPASSED) ||
-           (state == FLOW_STATE_CAPTURE_BYPASSED)) {
-        return;
-    }
-
-    FlowCounters *fc = SCCalloc(sizeof(FlowCounters), 1);
-    if (fc) {
-        FlowSetStorageById(p->flow, GetFlowBypassCounterID(), fc);
-    } else {
-        return;
+    if (p->flow) {
```

Comment:
can't we just bail early if p->flow == NULL? I assume we shouldn't run p->BypassPacketsFlow(p) on a packet w/o flow anyway?

Reply:
This patch suggests the same https://github.com/OISF/suricata/pull/3886/commits/8a08bbc217636aa4fe36d8a557cde4d0924ff68a

Reply:
No, we can run the callback function even if we don't have flow as the capture method may have a flow. This is the case for pf_ring and Netfilter.